File: key-value-search/perl/reverse_api_example.post.pl
#!/usr/bin/perl

use HTTP::Request::Common qw{ POST }; # From CPAN
use JSON qw( decode_json );           # From CPAN
use LWP::Protocol::https;             # From CPAN
use LWP::Simple;                      # From CPAN
use LWP::UserAgent;                   # From CPAN

use strict;
use warnings;

########################
# Fill in your details #
########################
my $user_name = 'Your reverse whois api username';
my $password = 'Your reverse whois api password';

my $url = 'https://www.whoisxmlapi.com/reverse-whois-api/search.php';

#######################
# Use a JSON resource #
#######################

my $responseJson = JSON->new->decode(reverseWhoisApiSearch('json'));
print "JSON\n---\n";
print JSON->new->pretty->encode($responseJson);

#######################
# Getting the data    #
#######################

sub reverseWhoisApiSearch {
    my $content = ' {
        "terms": [
            {
                "section": "Admin",
                "attribute": "organization",
                "value": "Wikimedia",
                "matchType": "anywhere",
                "exclude": false
            },
            {
                "section": "technical",
                "attribute": "city",
                "value": "San Francisco",
                "matchType": "anywhere",
                "exclude": false
            }
        ],
        "recordsCounter": false,
        "outputFormat": "json",
        "username": "' . $user_name . '",
        "password": "' . $password . '",
        "rows": 100
    }';

    my $ua = LWP::UserAgent->new(ssl_opts => { verify_hostname => 0 });
    my $req = HTTP::Request->new('POST', $url);

    $req->header('Content-Type' => 'application/json');
    $req->header('Accept', 'application/json');
    $req->content($content);

    my $response = $ua->request($req);
    print $response->content;
    return $response->content;
}

File: key-value-search/ruby/reverse_api_example_post.rb
require 'json'
require 'net/https'
require 'openssl'
require 'uri'
require 'yaml' # only needed to print the returned result in a very pretty way

########################
# Fill in your details #
########################
username = 'Your reverse whois api username'
password = 'Your reverse whois api password'

#######################
# Use a JSON resource #
#######################
format = 'json'
url = 'https://www.whoisxmlapi.com/reverse-whois-api/search.php'

content = {
  terms: [
    {
      section: 'Registrant',
      attribute: 'Name',
      matchType: 'BeginsWith',
      value: 'Mark',
      exclude: false
    },
    {
      section: 'Technical',
      attribute: 'Country',
      matchType: 'Anywhere',
      value: 'US',
      exclude: true
    }
  ],
  mode: 'preview',
  recordsCounter: false,
  username: username,
  password: password,
  outputFormat: format
}

uri = URI.parse(url)
http = Net::HTTP.new(uri.host, uri.port)

# Connect using ssl
http.use_ssl = true
http.verify_mode = OpenSSL::SSL::VERIFY_NONE
request = Net::HTTP::Post.new(uri.request_uri)

# Set headers
request.add_field('Content-Type', 'application/json')
request.add_field('Accept', 'application/json')
request.body = content.to_json

# Get response
response = http.request(request)

# Print parsed json
puts JSON.parse(response.body).to_yaml